src/test/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImplTest.scala

```diff
@@ -70,12 +70,14 @@ class OfferProcessorImplTest extends UnitTest {
     "match successful, launch tasks successful" in new Fixture {
       Given("an offer")
       val dummySource = new DummySource
-      val tasksWithSource = tasks.map(task => InstanceOpWithSource(dummySource, f.provision(task._1, task._2, task._3)))
+      val tasksWithSource = tasks.map {
+        case (taskInfo, task, instance) => InstanceOpWithSource(dummySource, f.provision(taskInfo, instance.instanceId, instance.agentInfo.get, instance.runSpecVersion, task, clock.now()))
+      }
 
       And("a cooperative offerMatcher and taskTracker")
       offerMatcher.matchOffer(offer) returns Future.successful(MatchedInstanceOps(offerId, tasksWithSource))
-      for (task <- tasks) {
-        val stateOp = InstanceUpdateOperation.Provision(task._3)
+      for ((_, _, instance) <- tasks) {
```

Comment:
You could use our new fancy [`Tasks`](https://github.com/mesosphere/marathon/pull/6687/files#diff-b113d7828c8c9e92a7004f07797e2e1aR38) extractor

```scala
tasks.foreach {
  case (_, _ , Instance(instanceId, Tasks(tasks), ... Some(agentInfo), runSpecVersion)) =>
}
```

Overall this seems very messy. Eg 
```scala
val dummySource = new DummySource
val tasksWithSource = tasks.map(...)
```

We really have to revisit this test.

Reply:
nice. But it's not merged yet though... Also I think this test has much bigger problems that not using the correct extractor 🙈 :D